app/src/lib/stores/app-store.ts

```diff
@@ -3347,44 +3348,89 @@ export class AppStore extends TypedBaseStore<IAppState> {
   public async _deleteBranch(
     repository: Repository,
     branch: Branch,
-    includeRemote: boolean
+    includeRemote?: boolean
   ): Promise<void> {
     return this.withAuthenticatingUser(repository, async (r, account) => {
-      const { branchesState } = this.repositoryStateCache.get(r)
-      let branchToCheckout = branchesState.defaultBranch
-
-      // If the default branch is null, use the most recent branch excluding the branch
-      // the branch to delete as the branch to checkout.
-      if (branchToCheckout === null) {
-        let i = 0
-
-        while (i < branchesState.recentBranches.length) {
-          if (branchesState.recentBranches[i].name !== branch.name) {
-            branchToCheckout = branchesState.recentBranches[i]
-            break
-          }
-          i++
-        }
-      }
+      const branchToCheckout = this.getBranchToCheckoutAfterDelete(branch, r)
+      const gitStore = this.gitStoreCache.get(r)
 
-      if (branchToCheckout === null) {
-        throw new Error(
-          `It's not possible to delete the only existing branch in a repository.`
+      if (branchToCheckout != null) {
```

Comment:
Shouldn't this be `!==`? I thought the linter would fail on these ones 🤔 
```suggestion
      if (branchToCheckout !== null) {
```